docs/content/concepts/io-management/io-managers.mdx

```diff
@@ -321,46 +327,57 @@ If you have specific requirements for where and how your outputs should be store
 
 To define an IO manager, use the <PyObject module="dagster" object="io_manager" displayText="@io_manager" /> decorator.
 
-```python
+```python file=/concepts/io_management/custom_io_manager.py startafter=start_io_manager_marker endbefore=end_io_manager_marker
+from dagster import IOManager, io_manager
+
+
 class MyIOManager(IOManager):
+    def _get_path(self, context) -> str:
+        return "/".join(context.asset_key.path)
+
     def handle_output(self, context, obj):
-        write_csv("some/path")
+        write_csv(self._get_path(context), obj)
 
     def load_input(self, context):
-        return read_csv("some/path")
+        return read_csv(self._get_path(context))
+
 
 @io_manager
-def my_io_manager(init_context):
+def my_io_manager():
     return MyIOManager()
 ```
 
 The <PyObject module="dagster" object="io_manager" /> decorator behaves nearly identically to the <PyObject module="dagster" object="resource" /> decorator. It yields an <PyObject module="dagster" object="IOManagerDefinition" />, which is a subclass of <PyObject module="dagster" object="ResourceDefinition" /> that will produce an <PyObject module="dagster" object="IOManager" />.
 
 The provided `context` argument for `handle_output` is an <PyObject module="dagster" object="OutputContext" />. The provided `context` argument for `load_input` is an <PyObject module="dagster" object="InputContext" />. The linked API documentation lists all the fields that are available on these objects.
 
-### Accessing the partition key
-
-IO managers interoperate smoothly with [partitions](/concepts/partitions-schedules-sensors/partitions). You can access the partition key for the current run using the `context` for both `load_input` and `handle_output`. If working with [assets](/concepts/assets/software-defined-assets), you can also access the asset-specific partition key or partition key range, though most of the time the run partition key will be equal to the asset partition key.
+### Handling partitioned assets
```

Comment:
A deliberate choice I made here was to focus entirely on partitioned _assets_. I think that most of the time someone writes a partitioned IO manager, they're using assets, and this allows us to avoid the confusing topic of the difference between partitioned assets and partitioned jobs.